packages/tests/shadydom/shady.html

```diff
@@ -1301,11 +1383,91 @@
+          ShadyDOM.wrap(el1).addEventListener('click', callback, true);
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
+          assert.equal(callback.callCount, 6, 'listener called 2x');
+          ShadyDOM.wrap(el1).addEventListener('click', callback, true);
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
+          assert.equal(callback.callCount, 8, 'listener called 2x');
+        });
+
+        test('listener removing de-duplicate based only on capture option', function () {
+          ShadyDOM.wrap(el1).addEventListener('click', callback);
+          ShadyDOM.wrap(el1).addEventListener('click', callback);
+          ShadyDOM.wrap(el1).removeEventListener('click', callback);
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
+          assert.equal(callback.callCount, 0, 'listener not called');
+          ShadyDOM.wrap(el1).addEventListener('click', callback, {
+            passive: true,
+          });
+          ShadyDOM.wrap(el1).removeEventListener('click', callback);
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
+          assert.equal(callback.callCount, 0, 'listener not called');
+          ShadyDOM.wrap(el1).addEventListener('click', callback);
+          ShadyDOM.wrap(el1).removeEventListener('click', callback, true);
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
+          assert.equal(callback.callCount, 1, 'listener called');
+          ShadyDOM.wrap(el1).removeEventListener('click', callback);
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
+          assert.equal(callback.callCount, 1, 'listener not called');
+          ShadyDOM.wrap(el1).addEventListener('click', callback, true);
+          ShadyDOM.wrap(el1).removeEventListener('click', callback);
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
+          assert.equal(callback.callCount, 2, 'listener called');
+          ShadyDOM.wrap(el1).removeEventListener('click', callback, true);
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
+          assert.equal(callback.callCount, 2, 'listener not called');
+        });
+
         test('support `once` option', function () {
           ShadyDOM.wrap(el1).addEventListener('click', callback, {once: true});
-          el1.click();
+          ShadyDOM.wrapIfNeeded(el1).dispatchEvent(
+            new Event('click', {bubbles: true, composed: true})
+          );
```

Comment:
We need to ensure the event dispatches through ShadyDOM. In `noPatch` mode it will not. In general we use `wrap` to ensure this, but the wrapper does not currently have `click` (a separate issue).

Reply:
Isn't `once` handled by `addEventListener` and the listener itself? I'm confused about which part a browser-dispatched event would miss.

Reply:
If the event is not dispatched via ShadyDOM, it won't get the `composed` flag on browsers that don't support it. We depend on that flag in the listeners.

Reply:
Oh, now I get it, so this change isn't specifically related to this test but rather for the others where there are polyfilled shadow roots?